.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,164 @@
+      - uses: release-drafter/release-drafter@v5
+        if: github.event_name == 'push'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  buildTest:
+    name: Build and test
+    runs-on: windows-latest 
+    needs: draftRelease
+    outputs:
+      buildNumber: ${{ steps.ver.outputs.BUILD_NUMBER }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: GetVersion
+        id: ver
+        shell: pwsh
+        run: |
+              if ("${{github.event.inputs.override}}")
+              {
+                Write-Host "Using manually entered override version"
+                $version = "${{ github.event.inputs.override}}"
+              }
+              else
+              {
+                $header = @{Authorization = 'Bearer ${{ secrets.GITHUB_TOKEN }}' }
+                $endPoint = "https://api.github.com/repos/${{github.repository}}/releases"
+                $info = Invoke-RestMethod -Uri $endPoint -Headers $header
+                $version = $info.name[0].Trim("v")
+              }
+              
+              Write-Host "::set-output name=BUILD_NUMBER::$($version)"
+          
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.101
+
+      - name: Install dependencies
+        run: dotnet restore Selenium.Axe/Selenium.Axe.sln
+        
+      - name: Update Firefox
+        run: choco upgrade firefox
+        
+      - name: Update Chrome
+        run: choco upgrade googlechrome
+        
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~\sonar\cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+          
+      - name: Cache SonarCloud scanner
+        id: cache-sonar-scanner
+        uses: actions/cache@v1
+        with:
+          path: .\.sonar\scanner
+          key: ${{ runner.os }}-sonar-scanner
+          restore-keys: ${{ runner.os }}-sonar-scanner
+          
+      - name: Install SonarCloud scanner
+        if: steps.cache-sonar-scanner.outputs.cache-hit != 'true'
+        shell: powershell
+        run: |
+          New-Item -Path .\.sonar\scanner -ItemType Directory
+          dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
+          
+      - name: Run tests
+        shell: powershell
+        run: dotnet test Selenium.Axe/Selenium.Axe.Test/Selenium.Axe.Test.csproj --logger trx --results-directory "${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CoverletOutput="${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CollectCoverage=true /p:CoverletOutputFormat=opencover -c Release -v D
```

Comment:
The results report only distinguishes the 3 target frameworks' worth of tests by their filenames. The [report documentation](https://github.com/dorny/test-reporter#supported-formats) suggests that, rather than using a `--results-directory` argument here, it might be preferable to set the result directory and/or result filename using `.csproj` properties so that `$(TargetFramework)` can be embedded into the file path; that would make it much easier to tell in the report which set of results is which. The docs suggest that the `.csproj` syntax is something like this (I haven't tested it):

```xml
<PropertyGroup>
  <VSTestLogger>trx%3bLogFileName=$(MSBuildProjectName).trx</VSTestLogger>
  <VSTestResultsDirectory>$(MSBuildThisFileDirectory)/TestResults/$(TargetFramework)</VSTestResultsDirectory>
</PropertyGroup>
```

Reply:
Updated with framework specific trx file names